src/lib/util/value.c

```diff
@@ -1096,37 +1100,38 @@ ssize_t fr_value_box_to_network(size_t *need, uint8_t *dst, size_t dst_len, fr_v
 	/*
 	 *	Fixed type would overflow output buffer
 	 */
-	if (max > dst_len) {
+	if (max > fr_dbuff_remaining(dbuff)) {
 		if (need) *need = max;
 		return 0;
 	}
 
 	switch (value->type) {
 	case FR_TYPE_IPV4_ADDR:
-		memcpy(dst, (uint8_t const *)&value->vb_ip.addr.v4.s_addr, sizeof(value->vb_ip.addr.v4.s_addr));
+		fr_dbuff_memcpy_in(dbuff, (uint8_t const *)&value->vb_ip.addr.v4.s_addr,
+				   sizeof(value->vb_ip.addr.v4.s_addr));
 		break;
 	/*
 	 *	Needs special mangling
 	 */
 	case FR_TYPE_IPV4_PREFIX:
-	 	dst[0] = value->vb_ip.prefix;
-		memcpy(&dst[1], (uint8_t const *)&value->vb_ip.addr.v4.s_addr,
-				 sizeof(value->vb_ip.addr.v4.s_addr));
+		fr_dbuff_memcpy_in(dbuff, &value->vb_ip.prefix, sizeof(value->vb_ip.prefix));
+		fr_dbuff_memcpy_in(dbuff, (uint8_t const *)&value->vb_ip.addr.v4.s_addr,
+				   sizeof(value->vb_ip.addr.v4.s_addr));
 		break;
 
 	case FR_TYPE_IPV6_ADDR:
-		if (value->vb_ip.scope_id > 0) *p++ = value->vb_ip.scope_id;
-		memcpy(p, value->vb_ip.addr.v6.s6_addr, sizeof(value->vb_ip.addr.v6.s6_addr));
+		if (value->vb_ip.scope_id > 0) fr_dbuff_memset(dbuff, value->vb_ip.scope_id, 1);
+		fr_dbuff_memcpy_in(dbuff, value->vb_ip.addr.v6.s6_addr, sizeof(value->vb_ip.addr.v6.s6_addr));
 		break;
 
 	case FR_TYPE_IPV6_PREFIX:
-		if (value->vb_ip.scope_id > 0) *p++ = value->vb_ip.scope_id;
-		*p++ = value->vb_ip.prefix;
-		memcpy(p, value->vb_ip.addr.v6.s6_addr, sizeof(value->vb_ip.addr.v6.s6_addr));
+		if (value->vb_ip.scope_id > 0) fr_dbuff_memset(dbuff, value->vb_ip.scope_id, 1);
```

Comment:
Purely because most people just associate memset with initialising memory, could you use fr_dbuff_bytes_in for this. i.e. `fr_dbuff_bytes_in(dbuff, value->vb_ip.scope_id)`